core/src/main/java/io/undertow/io/AsyncSenderImpl.java

```diff
@@ -408,11 +429,15 @@ private void invokeOnComplete() {
             this.buffer = null;
             this.fileChannel = null;
             this.callback = null;
-            inCallback = true;
+            writeThread = null;
+            inCallback = Thread.currentThread();
             try {
                 callback.onComplete(exchange, this);
             } finally {
-                inCallback = false;
+                inCallback = null;
+            }
+            if (Thread.currentThread() != writeThread) {
```

Comment:
L432 means this always evaluates as true.

Reply:
onComplete can do another write.

Reply:
Sorry for the noise!

Reply:
no need to be sorry, a review is always welcome!